testing/simulator/src/fallback_sim.rs

```diff
@@ -0,0 +1,248 @@
+            let beacon_nodes = if i == vc_count {
+                vec![i, 0]
+            } else {
+                vec![i, i + 1]
+            };
+            executor.spawn(
+                async move {
+                    let mut validator_config = testing_validator_config();
+                    if post_merge_sim {
+                        validator_config.fee_recipient = Some(SUGGESTED_FEE_RECIPIENT.into());
+                    }
+                    println!("Adding validator client {}", i);
+                    network_1
+                        .add_validator_client_with_fallbacks(
+                            validator_config,
+                            i,
+                            beacon_nodes,
+                            files,
+                        )
+                        .await
+                        .expect("should add validator");
+                },
+                "vc",
+            );
+        }
+
+        let duration_to_genesis = network.duration_to_genesis().await;
+        println!("Duration to genesis: {}", duration_to_genesis.as_secs());
+        sleep(duration_to_genesis).await;
+
+        if post_merge_sim {
+            let executor = executor.clone();
+            let network_2 = network.clone();
+            executor.spawn(
+                async move {
+                    println!("Mining pow blocks");
+                    let mut interval = tokio::time::interval(Duration::from_secs(seconds_per_slot));
+                    for i in 1..=TERMINAL_BLOCK + 1 {
+                        interval.tick().await;
+                        let _ = network_2.mine_pow_blocks(i);
+                    }
+                },
+                "pow_mining",
+            );
+        }
+        /*
+         * Start the checks that ensure the network performs as expected.
+         *
+         * We start these checks immediately after the validators have started. This means we're
+         * relying on the validator futures to all return immediately after genesis so that these
+         * tests start at the right time. Whilst this is works well for now, it's subject to
+         * breakage by changes to the VC.
+         */
+
+        let (disconnect, reconnect, check_attestations) = futures::join!(
+            checks::disconnect_from_execution_layer(
+                network.clone(),
+                Epoch::new(BELLATRIX_FORK_EPOCH),
+                slot_duration,
+                0
+            ),
+            checks::reconnect_to_execution_layer(
+                network.clone(),
+                Epoch::new(BELLATRIX_FORK_EPOCH),
+                slot_duration,
+                0,
+                2,
+            ),
+            checks::check_attestation_correctness(
+                network.clone(),
+                Epoch::new(0),
+                Epoch::new(END_EPOCH - 2),
+                MinimalEthSpec::slots_per_epoch(),
+                slot_duration,
+                1,
+            ),
+        );
```

Comment:
Why not write this with two futures

```rust
let test_sequence = async {
    wait_for(Epoch::new(BELLATRIX_FORK_EPOCH)).await
    disconnect_from_execution_layer().await
    wait_for(Epoch::new(BELLATRIX_FORK_EPOCH) + 2).await
    reconnect_to_execution_layer().await
}

futures::join!(
    test_sequence,
    checks::check_attestation_correctness(
```

Reply:
To decide the ACCEPTABLE_FALLBACK_ATTESTATION_HIT_PERCENTAGE = 95, what percentage do the disconnected node's validators represent of the total network? And for the disconnected 2 epochs what ratio represent of the full test duration?

Reply:
The `fallback-sim` currently runs  for 16 epochs and disconnection period is from epochs 1 to 15. The attestation performance is measured from epochs 0 to 14 (two epoch lag is required to check for late inclusions). All VCs are affect by the disconnection, that means that 13/14 (~93%) of epochs rely on the fallback mechanism.